x-pack/legacy/plugins/lens/public/indexpattern_plugin/field_item.tsx

```diff
@@ -24,7 +87,21 @@ function wrapOnDot(str?: string) {
   return str ? str.replace(/\./g, '.\u200B') : '';
 }
 
-export function FieldItem({ field, indexPattern, highlight, exists }: FieldItemProps) {
+export function FieldItem({
```

Comment:
This component feels a bit too complex now, and I think it would benefit from being broken into smaller pieces. Those pieces can still all live in this file, but it's currently pretty hard to follow all of the indentations, brackets, etc.

If I were to make a list of the things / responsibilities here, it would look something like this:

```js
// Selectors
function wrapOnDot(str);
function getFormatter(opts);

// Actions / effects
function loadFieldStats(opts);

// Components
function FieldItem(props); // The outer component
function FieldStats(props); // The flyout container
function HilightedText(props); // The hilighted text (currently lines 105-119)
function FieldStatsSwitcher(props); // Maybe the dropdown / chooser
function FieldStatsHistogram(props); 
function FieldStatsTopValues(props);
```